src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,156 @@
+package duke;
+
+// import duke packages
+import duke.command.Deadline;
+import duke.command.Event;
+import duke.command.Todo;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.Ui;
+
+// import java packages
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A chatbot that stores tasks given by a user.
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList cmdList;
+    private Ui ui;
+
+    /**
+     * Creates a Duke chatbot.
+     *
+     * @param filePath Filepath to save tasks and load tasks.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            cmdList = new TaskList(storage.loadTasks());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            cmdList = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the Duke chatbot.
+     */
+    public void run() {
+        ui.sayHello();
+        int count = 0 + cmdList.getLength();
+
+        // Create a scanner to read from standard input.
+        Scanner sc = new Scanner(System.in);
+        String cmd = sc.nextLine();
+
+        while (Parser.isNotBye(cmd)) {
+            try {
+                if (Parser.isList(cmd)) {       // when command given is "list"
```

Comment:
Should the comment be in a separate line from the code?
This is not explicitly stated in the Java coding standard but can help with consistency with the rest of the code.

Suggestion :
```java
if (Parser.isList(cmd)) {       
    // when command given is "list"
    ui.printList(cmdList);
```
I have noticed this issue a several times in this file as well.